### DISCUSSION TITLE
Include "BatchApex" in template options for `apex generate class`

### DISCUSSION THREAD

**Discussion by k-capehart** [forcedotcom/cli#3101] [Ideas]
For the `apex generate class` command the template options are: `ApexException | ApexUnitTest | BasicUnitTest | DefaultApexClass | InboundEmailService`.

I think it would be useful to include an option for an Apex class that implements the `Database.Batchable<sObject>` interface.

I believe we just need to include the template in [this repository](https://github.com/forcedotcom/salesforcedx-templates/tree/main/src/templates/apexclass), and then add the option in the CLI.

The template would look something like this:

```apex
public with sharing class <%= apiName %> implements Database.Batchable<sObject> {
  public <%= apiName %>() {

  }

  public Database.QueryLocator start(Database.BatchableContext bc) {

  }

  public void execute(Database.BatchableContext bc, List<sObject> scope) {

  }

  public void finish(Database.BatchableContext bc) {
  
  }
}
```

Is this something people think would be useful?

**Comment by klouf-threshold** [comment]
Agreed, although you can actually already do this locally: https://developer.salesforce.com/docs/platform/sfvscode-extensions/guide/byotemplate.html
